shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/DynamicFeatureManager.java

```diff
@@ -0,0 +1,152 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.dynamicfeatures;
+
+import io.flutter.embedding.engine.FlutterJNI;
+
+// TODO: add links to external documentation on how to use split aot features.
+/**
+ * Basic interface that handles downloading and loading of dynamic features.
+ *
+ * <p>Flutter dynamic feature support is still in early developer preview and should not be used in
+ * production apps yet.
+ *
+ * <p>The Flutter default implementation is PlayStoreDynamicFeatureManager.
+ *
+ * <p>DynamicFeatureManager handles the embedder/Android level tasks of downloading, installing, and
+ * loading Dart deferred libraries. A typical code-flow begins with a Dart call to loadLibrary() on
+ * deferred imported library. See https://dart.dev/guides/language/language-tour#deferred-loading
+ * This call retrieves a unique identifier called the loading unit id, which is assigned by
+ * gen_snapshot during compilation. The loading unit id is passed down through the engine and
+ * invokes downloadDynamicFeature. Once the feature module is downloaded, loadAssets and
+ * loadDartLibrary should be invoked. loadDartLibrary should find shared library .so files for the
+ * engine to open and pass the .so path to FlutterJNI.loadDartDeferredLibrary. loadAssets should
+ * typically ensure the new assets are available to the engine's asset manager by passing an updated
+ * Android AssetManager to the engine via FlutterJNI.updateAssetManager.
+ *
+ * <p>The loadAssets and loadDartLibrary methods are separated out because they may also be called
+ * manually via platform channel messages. A full downloadDynamicFeature implementation should call
+ * these two methods as needed.
+ *
+ * <p>A dynamic feature module is uniquely identified by a module name as defined in
+ * bundle_config.yaml. Each feature module may contain one or more loading units, uniquely
+ * identified by the loading unit ID and assets.
+ */
+public interface DynamicFeatureManager {
+  /**
+   * Sets the FlutterJNI to be used to communication with the Flutter native engine.
+   *
+   * <p>A FlutterJNI is required in order to properly execute loadAssets and loadDartLibrary.
+   *
+   * <p>Since this class may be instantiated for injection before the FlutterEngine and FlutterJNI
+   * is fully initialized, this method should be called to provide the FlutterJNI instance to use
+   * for use in loadDartLibrary and loadAssets.
+   */
+  public abstract void setJNI(FlutterJNI flutterJNI);
```

Comment:
yak creep / optional: one way you can get around this is letting the FlutterJNI be in the injector too. Then you can just read from the injector right before use.